planet/api/experimental/basemaps_client.py

```diff
@@ -22,33 +23,42 @@ def __init__(self, api_key=None, base_url=BASE_URL):
         session.mount("https://", adapter)
         self.session = session
 
-    def _get(self, url):
-        resp = self.session.get(url)
+    def _get(self, url, **kwargs):
+        resp = self.session.get(url, params=kwargs)
         if not resp.ok:
             raise APIException('{}: {}'.format(resp.status_code, resp.content))
         return resp
 
-    def list_basemap_series(self):
+    def _post(self, url, body):
+        resp = self.session.post(url, json=body)
+        if not resp.ok:
+            raise APIException('{}: {}'.format(resp.status_code, resp.content))
+        return resp
+
+    def list_mosaics(self, name__is=None, name__contains=None):
```

Comment:
As implemented, this will only return the first 50 (default page size) mosaics matching the provided filters.

Reply:
This generator pattern will crawl through the `_next` links until there are no more mosaics to fetch:
```python
def _get_next_url(resp):
    return resp.json().get('_links', {}).get('_next')

def list_mosaics(self, name__is=None, name__contains=None):
    # yield one mosaic at a time from the first page of results
    url = self.base_url + 'basemaps/v1/mosaics/'
    resp = self._get(url, name__is=name__is, name__contains=name__contains)
    mosaics = resp.json().get('mosaics', [])
    for mosaic in mosaics:
        yield mosaic
    # crawl through next link results, yielding mosaics until there are none left
    next_url = _get_next_url(resp)
    while next_url:
        # no extra params needed, next_url has querystring already
        resp = self._get(next_url)
        mosaics = resp.json().get('mosaics', [])
        for mosaic in mosaics:
            yield mosaic
        next_url = _get_next_url(resp)
```